### PR TITLE
Add support to remote uris in service control scripts in Windows and Linux

### PR DESCRIPTION
**Pull request by rapphil**
**Description:** It is now possible to pass remote uris such as http, https and s3 to the control scripts in Windows an Linux.

**Testing:** Tested changes locally

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tools/ctl/linux/aws-otel-collector-ctl [hunk 1]
```diff
@@ -19,16 +19,16 @@ set -u
 readonly AGENTDIR="/opt/aws/aws-otel-collector"
 readonly CMDDIR="${AGENTDIR}/bin"
 readonly CONFDIR="${AGENTDIR}/etc"
+readonly ENV_FILE="${CONFDIR}/.env"
 readonly DFT_CONFDIR="${AGENTDIR}/var"
 readonly LOGDIR="${AGENTDIR}/logs"
-readonly RESTART_FILE="${CONFDIR}/restart"
```
Comment (bryan-aguilar) [inline tools/ctl/linux/aws-otel-collector-ctl:24]: Why was this removed?

File: tools/ctl/linux/aws-otel-collector-ctl [hunk 2]
```diff
@@ -44,20 +44,41 @@ UsageString="
   status:                                 get the status of the agent process.
 
   -c: configuration
-  file:<file-path>:                       file path on the host
+  <config-uri>:                           - file path on the host. E.g.: /tmp/config.yaml
```
Comment (bryan-aguilar) [inline tools/ctl/linux/aws-otel-collector-ctl:47]:
```suggestion
  <config-uri>:                           - file path on the host. E.g.: file:/tmp/config.yaml
```
Is the conf map provider required for file uris still?

File: tools/ctl/linux/aws-otel-collector-ctl [hunk 3]
```diff
@@ -44,20 +44,41 @@ UsageString="
...
     fi
+}
+
+aoc_start() {
+    config="${1:-}"
+
+    if [[ $config =~ ^(https|http|s3)://.* ]]; then
+        aoc_config_remote_uri $config
+    else
+        aoc_config_local_uri $config
```
Comment (bryan-aguilar) [inline tools/ctl/linux/aws-otel-collector-ctl:80]: Why not use the `file:` [config map provider](https://github.com/open-telemetry/opentelemetry-collector/blob/main/confmap/provider/fileprovider/provider.go) instead of copying?

**Review by bryan-aguilar — COMMENTED** [review]
This may be scope creeping this PR but how can end users provide multiple configurations to the collector with this implementation?

File: tools/ctl/linux/aws-otel-collector-ctl (hunk 3 above)
Comment (rapphil) [reply tools/ctl/linux/aws-otel-collector-ctl:80]: because I don't want to change the previous behavior. customers might be copying files themselves directly to that location and just restarting the service.

File: tools/ctl/linux/aws-otel-collector-ctl (hunk 1 above)
Comment (rapphil) [reply tools/ctl/linux/aws-otel-collector-ctl:24]: it was unused.

File: tools/ctl/linux/aws-otel-collector-ctl (hunk 2 above)
Comment (rapphil) [reply tools/ctl/linux/aws-otel-collector-ctl:47]: no. that is optional.

**Comment by rapphil** [comment]
> This may be scope creeping this PR but how can end users provide multiple configurations to the collector with this implementation?

@bryan-aguilar we don't support multiple configs today anywhere. I think this can be a separated issue no?

**Comment by bryan-aguilar** [comment]
> > This may be scope creeping this PR but how can end users provide multiple configurations to the collector with this implementation?
> 
> @bryan-aguilar we don't support multiple configs today anywhere. I think this can be a separated issue no?

I think this is true for everything but container images. I think you could provide multiple configs to a container image command.

File: tools/ctl/linux/aws-otel-collector-ctl [hunk 4]
```diff
@@ -44,20 +44,41 @@ UsageString="
...
+                                          - https uri. E.g.: https://example.com/config
+                                          - s3 uri. E.g.: s3://bucket/config
 
   "
 
-aoc_start() {
+aoc_config_remote_uri() {
     config="${1:-}"
 
+    echo "config=\"--config ${config}\"" > $ENV_FILE
```
Comment (Aneurysm9) [inline tools/ctl/linux/aws-otel-collector-ctl:57]: This will overwrite the entire `$ENV_FILE`.  Is that desirable?

File: tools/ctl/linux/aws-otel-collector-ctl [hunk 5]
```diff
@@ -44,20 +44,41 @@ UsageString="
...
 
     if [ ! -f $CONFDIR/config.yaml ]; then
         cp $DFT_CONFDIR/.config.yaml $CONFDIR/config.yaml
     fi
+}
+
+aoc_start() {
+    config="${1:-}"
+
+    if [[ $config =~ ^(https|http|s3)://.* ]]; then
```
Comment (Aneurysm9) [inline tools/ctl/linux/aws-otel-collector-ctl:77]: This seems brittle.  We'll have to change this regex every time a new config provider is added.

File: tools/ctl/linux/aws-otel-collector-ctl (hunk 4 above)
Comment (vasireddy99) [reply tools/ctl/linux/aws-otel-collector-ctl:57]: can we use >> instead?

Comment (Aneurysm9) [reply tools/ctl/linux/aws-otel-collector-ctl:57]: Appending will result in an ever-growing file setting the same variable many times.  Either this value shouldn't come from that file or this will need logic to detect whether the value is already being set in the file and react accordingly.

Comment (rapphil) [reply tools/ctl/linux/aws-otel-collector-ctl:57]: > This will overwrite the entire $ENV_FILE. Is that desirable?

Yes, that is the intention.

File: tools/ctl/linux/aws-otel-collector-ctl (hunk 5 above)
Comment (rapphil) [reply tools/ctl/linux/aws-otel-collector-ctl:77]: I will replace by a more relaxed regex.

**Review by bryan-aguilar — APPROVED** [review]
